src/Model.php

```diff
@@ -39,6 +39,44 @@ class Model implements \ArrayAccess, \IteratorAggregate
     use CollectionTrait;
     use ReadableCaptionTrait;
 
+    /** @const string */
+    public const HOOK_BEFORE_LOAD = 'beforeLoad';
+    /** @const string */
+    public const HOOK_AFTER_LOAD = 'afterLoad';
+    /** @const string */
+    public const HOOK_BEFORE_UNLOAD = 'beforeUnload';
+    /** @const string */
+    public const HOOK_AFTER_UNLOAD = 'afterUnload';
+
+    /** @const string */
+    public const HOOK_BEFORE_INSERT = 'beforeInsert';
+    /** @const string */
+    public const HOOK_AFTER_INSERT = 'afterInsert';
+    /** @const string */
+    public const HOOK_BEFORE_UPDATE = 'beforeUpdate';
+    /** @const string */
+    public const HOOK_AFTER_UPDATE = 'afterUpdate';
+    /** @const string */
+    public const HOOK_BEFORE_DELETE = 'beforeDelete';
+    /** @const string */
+    public const HOOK_AFTER_DELETE = 'afterDelete';
+
+    /** @const string */
+    public const HOOK_BEFORE_SAVE = 'beforeSave';
+    /** @const string */
+    public const HOOK_AFTER_SAVE = 'afterSave';
+
+    /** @const string Executed when execution of self::atomic() failed. */
+    public const HOOK_ROLLBACK = self::class . '@rollback';
```

Comment:
not sure it this helps

Reply:
it will be also better to disable StyleCI. Everywhere

Reply:
fixed.